src/vs/workbench/browser/actions/toggleTabsVisibility.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import { TPromise } from 'vs/base/common/winjs.base';
+import nls = require('vs/nls');
+import { Registry } from 'vs/platform/registry/common/platform';
+import { Action } from 'vs/base/common/actions';
+import { SyncActionDescriptor } from 'vs/platform/actions/common/actions';
+import { IWorkbenchActionRegistry, Extensions } from 'vs/workbench/common/actions';
+import { IConfigurationService, ConfigurationTarget } from 'vs/platform/configuration/common/configuration';
+import { KeyCode, KeyMod } from 'vs/base/common/keyCodes';
+
+export class ToggleTabsVisibilityAction extends Action {
+
+	public static ID = 'workbench.action.toggleTabsVisibility';
+	public static LABEL = nls.localize('toggleTabs', "Toggle Tab Visibility");
+
+	private static tabsVisibleKey = 'workbench.editor.showTabs';
+
+	constructor(
+		id: string,
+		label: string,
+		@IConfigurationService private configurationService: IConfigurationService
+	) {
+		super(id, label);
+	}
+
+	public run(): TPromise<any> {
+		const visibility = this.configurationService.getValue<string>(ToggleTabsVisibilityAction.tabsVisibleKey);
+		const newVisibilityValue = !visibility;
+
+		return this.configurationService.updateValue(ToggleTabsVisibilityAction.tabsVisibleKey, newVisibilityValue, ConfigurationTarget.USER);
```

Comment:
@sandy081 how is the state of things here, shouldn't we stay away from setting an explicit Configuration.TARGET and rather let the service decide where to update the value based on where it is defined?

@maxfurman the issue here is that this setting could be defined on a workspace level and then writing it back to user level would not cause any visual change.

Reply:
@bpasero Yes, Do not use the target, if the action is intended to change the current effective value. Current state is that all existing actions are still using the target (debt item). Make sure new ones do not cause the debt.

Reply:
@maxfurman can you make that change to not set the target?

Reply:
No problem. Simply removing the third argument to updateValue will do the trick, correct?